### PR TITLE
Add some Armv8-M assembly routines

### PR DESCRIPTION
**Pull request by hug-dev**
Adds access to `MSP_NS` and the `BXNS` instruction.
Also adds `__dsb` which was missing.

Executed `cargo xtask assemble` and pushed everything, I hope that's enough.

### CODE REVIEW

File: src/register/msp.rs [hunk 1]
```diff
@@ -11,3 +11,19 @@ pub fn read() -> u32 {
 pub unsafe fn write(bits: u32) {
     call_asm!(__msp_w(bits: u32));
 }
+
+/// Reads the Non-Secure CPU register from Secure state.
+/// Executing this function in Non-Secure state will return zeroes.
```
Comment (jonas-schievink) [inline src/register/msp.rs:16]:
```suggestion
///
/// Executing this function in Non-Secure state will return zeroes.
```

File: src/register/msp.rs [hunk 2]
```diff
@@ -11,3 +11,19 @@ pub fn read() -> u32 {
...
+#[cfg(armv8m)]
+#[inline]
+pub fn read_ns() -> u32 {
+    call_asm!(__msp_ns_r() -> u32)
+}
+
+/// Writes `bits` to the Non-Secure CPU register from Secure state.
+/// Executing this function in Non-Secure state will be ignored.
```
Comment (jonas-schievink) [inline src/register/msp.rs:24]:
```suggestion
///
/// Executing this function in Non-Secure state will be ignored.
```

**Review by jonas-schievink — COMMENTED** [review]
LGTM, added a small rustdoc improvement

**Comment by hug-dev** [comment]
Done both :)

**Review by jonas-schievink — APPROVED** [review]
bors r+

**Pull request merged**